tests/test_pex_bootstrapper.py

```diff
@@ -34,39 +34,25 @@ def test_get_pex_info():
 
 @pytest.mark.skipif(IS_PYPY)
 def test_find_compatible_interpreters():
-  pex_python_path = ':'.join([
-    ensure_python_interpreter('2.7.9'),
-    ensure_python_interpreter('2.7.10'),
-    ensure_python_interpreter('2.7.11'),
-    ensure_python_interpreter('3.4.2'),
-    ensure_python_interpreter('3.5.4'),
-    ensure_python_interpreter('3.6.2'),
-    ensure_python_interpreter('3.6.3')
-  ])
+  py27 = ensure_python_interpreter(PY27)
+  py35 = ensure_python_interpreter(PY35)
+  py36 = ensure_python_interpreter(PY36)
+  pex_python_path = ':'.join([py27, py35, py36])
 
-  interpreters = find_compatible_interpreters(pex_python_path, ['>3'])
-  assert interpreters[0].binary == pex_python_path.split(':')[3]  # 3.4.2
+  def find_interpreters(*constraints):
+    return [interp.binary for interp in find_compatible_interpreters(pex_python_path, constraints)]
 
-  interpreters = find_compatible_interpreters(pex_python_path, ['<3'])
-  assert interpreters[0].binary == pex_python_path.split(':')[0]  # 2.7.9
+  assert [py35, py36] == find_interpreters('>3')
+  assert [py27] == find_interpreters('<3')
 
-  interpreters = find_compatible_interpreters(pex_python_path, ['>3.5.4'])
-  assert interpreters[0].binary == pex_python_path.split(':')[5]  # 3.6.2
+  assert [py36] == find_interpreters('>{}'.format(PY35))
```

Comment:
Part of the reason I made these tests at this level of granularity was to assure that patch-level version constraints between constraints with the same major versions are respected, however I would imagine that is covered in testing elsewhere so this change is fine with me.

Reply:
Understood. Yeah, the underlying comparison is done by `PythonIdentity.matches` which is actually not unit-tested, but it could and should be. The testing here was just way too expensive.